src/SpectatorView.Unity/Assets/SpectatorView/Scripts/Socketer/TCPNetworkConnection.cs

```diff
@@ -86,15 +86,15 @@ public void Send(ref byte[] data)
             {
                 try
                 {
-                    socketerClient.SendNetworkMessage(data, sourceId);
+                    var payload = new byte[length];
+                    Array.Copy(data, offset, payload, 0, length);
+                    socketerClient.SendNetworkMessage(payload, sourceId);
```

Comment:
Beyond the scope of this review, but are we sure all the small writes are getting coalesced and we're not spamming lots of packets? A quick look in a packet sniffer could be useful.